src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/Migrations/20240628135434_EES5235_GrantSequencePrivileges.cs

```diff
@@ -12,14 +13,19 @@ public partial class EES5235_GrantSequencePrivileges : Migration
     /// <inheritdoc />
     protected override void Up(MigrationBuilder migrationBuilder)
     {
+        var isLocalEnvironment = 
+            Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == Environments.Development;
```

Comment:
Well found - I struggled to work out how to determine if you're in development or not when away from the host builder code!

Reply:
I've added this to `MigrationBuilderExtensions` now so that within a migration you can do `migrationBuilder.IsLocalEnvironment()`.

Reply:
I've added this to `MigrationBuilderExtensions` now so that within a migration you can do something like `migrationBuilder.IsEnvironment(Environments.Development)` or `migrationBuilder.IsEnvironment(Environments.Production)`.

The reason why we check `migrationBuilder.IsEnvironment(Environments.Production)` rather than `!migrationBuilder.IsEnvironment(Environments.Development)`  is that we don't want to run `ALTER DEFAULT PRIVILEGES` in integration tests because that role doesn't exist. In the local environment it's been run by the Docker entrypoint init script.

I wasn't sure if we should be doing something in the integration tests before running the database migration to run the init script, or even going as far as to make sure each integration test migrates the database with the same user setup that it would do in a production and then runs the test as the same user that it would do in production, which might be a different user. That seems like it might be worthwhile but not an easy change to make. Using a pre-existing image that already has a migrated database seems like it would be better for performance anyway. I have left this for now.

Reply:
Makes sense, thanks!